### PR TITLE
Do not expose absolute addresses in ModuleData

### PR DESCRIPTION
**Pull request by florian-kuebler**
ModuleData exists in ModuleManager once per path and build id.
So if a module is loaded twice at two different locations
in memory, there will be only one instance of ModuleData.

Thus, it must not expose any information about where it
is loaded in memory. For this, ProcessData should be
used.

Test: Check manaual instrumentation still works on OrbitTest
Bug: http://b/236710308

### CODE REVIEW

File: src/FakeClient/FakeClientMain.cpp [hunk 1]
```diff
@@ -377,6 +381,7 @@ int main(int argc, char* argv[]) {
     ErrorMessageOr<std::vector<orbit_grpc_protos::ModuleInfo>> modules_or_error =
         orbit_module_utils::ReadModules(options.process_id);
     ORBIT_FAIL_IF(modules_or_error.has_error(), "%s", modules_or_error.error().message());
+    process.UpdateModuleInfos(modules_or_error.value());
```
Comment (dpallotti) [inline src/FakeClient/FakeClientMain.cpp:384]: If `!options.enable_api && !calculate_frame_time`, we are passing a process without modules to `Capture`, which is fine but feels a tad wrong. Should we just move this part out of the `if` to the initialization of `process`? ...

File: src/FakeClient/FakeClientMain.cpp [hunk 2]
```diff
@@ -398,6 +403,7 @@ int main(int argc, char* argv[]) {
...
```
Comment (dpallotti) [inline src/FakeClient/FakeClientMain.cpp:406]: ... And then also remove this part? We actually get rid of code duplication this way.

File: src/FakeClient/FakeClientMain.cpp (hunk 1 above)
Comment (dpallotti) [inline src/FakeClient/FakeClientMain.cpp:384]: I'm a bit scared of making a change to `FakeClient` without any manual testing. Would you be able to do that?

**Review by dpallotti — COMMENTED** [review]
Thanks a lot for this. Mainly a comment regarding a quick test of the changes to the fake client.

File: src/CaptureClient/CaptureClient.cpp [hunk 3]
```diff
@@ -92,7 +100,8 @@ std::vector<ApiFunction> FindApiFunctions(const orbit_client_data::ModuleManager
 }
 
 [[nodiscard]] orbit_grpc_protos::CaptureOptions ToGrpcCaptureOptions(
-    const ClientCaptureOptions& options, const orbit_client_data::ModuleManager& module_manager) {
+    const ClientCaptureOptions& options, const orbit_client_data::ModuleManager& module_manager,
+    const orbit_client_data::ProcessData& process) {
```
Comment (pierricgimmig) [inline src/CaptureClient/CaptureClient.cpp:104]: nit. I would rename `process` -> `process_data` to give a better indication of the type without having to look at the signature. Also elsewhere in the PR.

**Review by pierricgimmig — APPROVED** [review]
Thanks for this @florian-kuebler, please see minor comment, but lgtm in general.

File: src/FakeClient/FakeClientMain.cpp (hunk 1 above)
Comment (dpallotti) [reply src/FakeClient/FakeClientMain.cpp:384]: Tested together offline.

**Review by dpallotti — APPROVED** [review]

**Pull request merged**